exp/api/v1beta1/azuremachinepool_default_test.go

```diff
@@ -328,6 +326,7 @@ func TestAzureMachinePool_SetNetworkInterfacesDefaults(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+		    g := NewWithT(t)
```

Comment:
This looks exactly right! I have a feeling there are a few more tests like this throughout the repo that could use a similar change but don't remember exactly where they are. It will be tedious, but I'm not sure there's a better way to find them than to search for every occurrence of `NewWithT` and `NewGomegaWithT` throughout the repo.